x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable.tsx

```diff
@@ -445,6 +481,13 @@ export class Embeddable
 
   destroy() {
     super.destroy();
+    if (this.inputReloadSubscriptions.length > 0) {
+      this.inputReloadSubscriptions.forEach((reloadSub) => {
+        reloadSub.unsubscribe();
+      });
+      // @ts-expect-error
+      this.input = null;
```

Comment:
I tested without this line and it worked as well. Is the unsubscribing enough? And great catch by the way!

Reply:
i think it works in most cases with those unsubs, but if you go back-forth couple of times, without this it still breaks.